src/main/java/org/sciborgs1155/robot/drive/Drive.java

```diff
@@ -1,263 +1,194 @@
+        s.setIdleMode(IdleMode.kBrake);
+        s.setOpenLoopRampRate(0);
+        s.setSmartCurrentLimit(50);
+    });
+
+  private final CANSparkMax bLeftMotor = 
+  SparkUtils.create(
+    BL_DRIVE_PORT,
+    s -> {
+        s.setInverted(false);
+        s.setIdleMode(IdleMode.kBrake);
+        s.setOpenLoopRampRate(0);
+        s.setSmartCurrentLimit(50);
+    });
+
+  private final CANSparkMax[] rightSparks = {fRightMotor, bRightMotor};
+  private final CANSparkMax[] leftSparks = {fLeftMotor, bLeftMotor};
+
+  private final MotorControllerGroup rightMotors = new MotorControllerGroup(rightSparks);
+  private final MotorControllerGroup leftMotors = new MotorControllerGroup(leftSparks);
+
+  // Gyros
+  @Log private final WPI_PigeonIMU gyro = new WPI_PigeonIMU(DrivePorts.PIGEON);
+  //Right and Left side encoders
+  private final RelativeEncoder rightEncoder = fRightMotor.getEncoder();
+  private final RelativeEncoder leftEncoder = fLeftMotor.getEncoder();
+  
+  @Log private final PIDController RdrivePID = new PIDController(kP, kI, kD);
+  @Log private final PIDController LdrivePID = new PIDController(kP,  kI, kD);
+
+  @Log private final SimpleMotorFeedforward RdriveFF = new SimpleMotorFeedforward(kS, kV, kA);
+  @Log private final SimpleMotorFeedforward LdriveFF = new SimpleMotorFeedforward(kS, kV, kA);
+
+  @Log private final DifferentialDrive drive = new DifferentialDrive(fLeftMotor, fRightMotor);
+
+  
 
   /**
-   * Returns the currently-estimated pose of the robot.
-   *
-   * @return The pose.
+   * Encoders PID and FF controllers, -> mostly done by now, maybe there is something else about this that can be worked on later
+   * DifferentialDriveOdometry -> used DifferentialDrivePoseEstimator instead rn
+   * Getters for pose, encoder values
+   * Setters for setpoints and other important values
+   * Style
    */
-  public Pose2d getPose() {
-    return odometry.getEstimatedPosition();
-  }
+  
 
-  /**
-   * Returns the heading of the robot, based on our pigeon
-   *
-   * @return A Rotation2d of our angle
-   */
-  public Rotation2d getHeading() {
-    return imu.getRotation2d();
-  }
+  //sets up Odometry
+  private final DifferentialDriveKinematics kinematics = new DifferentialDriveKinematics(ROBOT_TRACK);
+  DifferentialDriveWheelSpeeds speeds = new DifferentialDriveWheelSpeeds(0, 0);
 
-  /**
-   * Resets the odometry to the specified pose.
-   *
-   * @param pose The pose to which to set the odometry.
-   */
-  public void resetOdometry(Pose2d pose) {
-    odometry.resetPosition(getHeading(), getModulePositions(), pose);
-  }
+  @Log private final DifferentialDrivePoseEstimator odometry = 
+    new DifferentialDrivePoseEstimator(
+      kinematics,
+      gyro.getRotation2d(), 
+      leftEncoder.getPosition(), 
+      rightEncoder.getPosition(), 
+      new Pose2d(),
+          VecBuilder.fill(0.05, 0.05, Units.degreesToRadians(5)), //these could probably be adjusted later, taken from wpilib docs
```

Comment:
Are you guys planning to use vision for odometry? If not, use DifferentialDriveOdometry instead of DifferentialDrivePoseEstimator. And if you are going to use vision for odometry, make sure you actually understand what these vectors do.